### PR TITLE
combined validation error set and update logic

### PR DESCRIPTION
**Pull request by bender2k14**
I found these two pattern matches on `VmBinding<'model, 'msg>` that were almost identical.  This PR extracts the common code.

The main difference was the model (`initialModel` vs `currentModel`), which is now passed in to the new function.

The other difference was what to do when `ValueNone` is matched.  The update case called `removeError name` while the initial case did nothing.  I unified this by having the initial case now also call `removeError name` since `removeError` will also do nothing after not finding `name` in the `errors` dictionary.

In addition to the extracted function being called, this adds one call to `Dictionary.Remove` and one `if` check per `twoWayValidate` initialization.  Probably fine.

I am looking ahead to #151.  I expect this change to help a bit since this decreases the number of pattern matches on `VmBinding<'model, 'msg>`.

### CODE REVIEW

File: src/Elmish.WPF/ViewModel.fs [hunk 1]
```diff
@@ -152,6 +152,16 @@ and [<AllowNullLiteral>] internal ViewModel<'model, 'msg>
...
 
+  let updateValidationError name model = function
+    | TwoWayValidate { Validate = validate } ->
+        match validate model with
+        | ValueNone -> removeError name
+        | ValueSome error -> setError error name
+    | _ -> ()
+
+  let setInitialValidationError name binding = updateValidationError name initialModel binding
+  let updateValidationError name binding = updateValidationError name currentModel binding
```
Comment (cmeeren) [inline src/Elmish.WPF/ViewModel.fs]: Are these necessary? Since they're only used once each, I think it's more clear if they are removed.

Comment (bender2k14) [reply src/Elmish.WPF/ViewModel.fs]: Nope.  I removed them and swapped the order of the `model` and `name` arguments.  I merged this change with the previous commit and force pushed.

**Comment by cmeeren** [comment]
Thanks! Go ahead and merge whenever you're ready.

**Comment by bender2k14** [comment]
> The main difference was the model (`initialModel` vs `currentModel`), which is now passed in to the new function.

In fact, it suffices to always use `currentModel` since `currentModel` is immediately initiated with `initialModel` and there is no mutation yet.  I don't mind leaving it as is though.

**Comment by cmeeren** [comment]
> In fact, it suffices to always use `currentModel` since `currentModel` is immediately initiated with `initialModel` and there is no mutation yet. I don't mind leaving it as is though.

Ah, noted. I don't care much either way. Feel free to change it if you want, or leave it as it is.

**Pull request merged**